src/CollisionHandler.cpp

```diff
@@ -21,25 +23,24 @@ sf::Vector2f CollisionHandler::handleTileCollision(const sf::FloatRect & entityA
 			//Determine direction in which to move entity
 			if (xDifference > 0.0f)
 			{
-				newPos.x = intersection.width;
+				movement.x += intersection.width;
 			}
 			else
 			{
-				newPos.x = -intersection.width;
+				movement.x -= intersection.width;
 			}
 		}
 		else
 		{
 			if (yDifference > 0.0f)
 			{
-				newPos.y = intersection.height;
+				movement.y += intersection.height;
 			}
 			else
 			{
-				newPos.y = -intersection.height;
+				movement.y -= intersection.height;
 			}
 		}
+		break;
```

Comment:
Couldn't get the code you suggested working properly. It caused the player to collide with every section of the wall instead of the part it was running into.